apps/roam/src/utils/getAllDiscourseNodesSince.ts

```diff
@@ -1,48 +1,171 @@
-import getDiscourseNodes from "./getDiscourseNodes";
+/* eslint-disable @typescript-eslint/naming-convention */
+import getDiscourseNodes, { type DiscourseNode } from "./getDiscourseNodes";
 import findDiscourseNode from "./findDiscourseNode";
+import { type OnloadArgs } from "roamjs-components/types";
+import getDiscourseNodeFormatExpression from "./getDiscourseNodeFormatExpression";
 
-type RoamDiscourseNodeData = {
+type ISODateString = string;
+
+export type RoamDiscourseNodeData = {
   author_local_id: string;
+  author_name: string;
   source_local_id: string;
   created: string;
   last_modified: string;
-  author_name: string;
   text: string;
+  type: string;
+  node_title?: string;
 };
 
-type ISODateString = string;
+export type DiscourseNodesSinceResult = {
+  pageNodes: RoamDiscourseNodeData[];
+  blockNodes: RoamDiscourseNodeData[];
+};
+
+export const getDiscourseNodeTypeWithSettingsBlockNodes = (
+  node: DiscourseNode,
+  sinceMs: number,
+  extensionAPI: OnloadArgs["extensionAPI"],
+): RoamDiscourseNodeData[] => {
+  const settingsKey = `discourse-graph-node-rule-${node.type}`;
+  const settings = extensionAPI.settings.get(settingsKey) as {
+    embeddingRef: string;
+    isFirstChild: boolean;
+  };
+  const regex = getDiscourseNodeFormatExpression(node.format);
+  const regexPattern = regex.source.replace(/\\/g, "\\\\").replace(/"/g, '\\"');
+  const firstChildUid =
+    settings.embeddingRef?.match(/\(\((.*?)\)\)/)?.[1] ?? "";
+  const queryBlock = `[
+      :find ?childString ?nodeUid        ?nodeCreateTime ?nodeEditTime ?author_local_id ?type ?author_name ?node-title
+      :keys text         source_local_id created         last_modified author_local_id   type author_name  node_title
+      :in $ ?firstChildUid ?type ?since
+      :where
+        [(re-pattern "${regexPattern}") ?title-regex]
+        [?node :node/title ?node-title]
+        [(re-find ?title-regex ?node-title)]
+        [?node :block/uid ?nodeUid]
+        [?node :create/time ?nodeCreateTime]
+        [?node :edit/time ?nodeEditTime]
+        [?s :block/uid ?firstChildUid]
+        [?s :block/string ?firstChildString]
+        [?bg :block/page ?node]
+        [?bg :block/string ?firstChildString]
+        [?bg :block/children ?child]
+        [?child :block/order 0]
+        [?child :block/string ?childString]
+        [?child :edit/time ?childEditTime]
+        [?child :create/user ?user-eid]
+        [?user-eid :user/uid ?author_local_id]
+        [?child :edit/user ?eu]
+        [?eu :user/display-name ?author_name]
+        [or 
+         [(> ?childEditTime ?since)]
+         [(> ?nodeEditTime ?since)]]
+        ]`;
+
+  const blockNode = window.roamAlphaAPI.data.q(
```

Comment:
@sid597 This should probably be `.backend`.   We can discuss at our next meeting.